File: .cursor/mcp/run_notion_mcp.mjs
import { existsSync, readFileSync } from "node:fs";
import { dirname, resolve } from "node:path";
import { fileURLToPath } from "node:url";
import { spawn } from "node:child_process";

function loadEnvFile(envPath) {
  if (!existsSync(envPath)) return;
  const lines = readFileSync(envPath, "utf8").split(/\r?\n/);
  for (const raw of lines) {
    const line = raw.trim();
    if (!line || line.startsWith("#")) continue;
    const idx = line.indexOf("=");
    if (idx <= 0) continue;
    const key = line.slice(0, idx).trim();
    let value = line.slice(idx + 1).trim();
    if ((value.startsWith('"') && value.endsWith('"')) || (value.startsWith("'") && value.endsWith("'"))) {
      value = value.slice(1, -1);
    }
    if (key && !process.env[key]) {
      process.env[key] = value;
    }
  }
}

const scriptDir = dirname(fileURLToPath(import.meta.url));
const envPath = resolve(scriptDir, "notion.env");
loadEnvFile(envPath);

if (!process.env.NOTION_TOKEN || !process.env.NOTION_TOKEN.trim()) {
  process.stderr.write(
    "[notion-mcp] Missing NOTION_TOKEN. Please copy .cursor/mcp/notion.env.example to .cursor/mcp/notion.env and fill it.\n",
  );
  process.exit(1);
}

function resolveNpx() {
  if (process.platform === "win32") {
    const defaultNpx = "C:\\Program Files\\nodejs\\npx.cmd";
    if (existsSync(defaultNpx)) return defaultNpx;
  }
  return "npx";
}

const npxCmd = resolveNpx();
const isWinCmd = process.platform === "win32" && npxCmd.toLowerCase().endsWith(".cmd");
const child = isWinCmd
  ? spawn(`"${npxCmd}"`, ["-y", "@notionhq/notion-mcp-server"], {
      cwd: scriptDir,
      stdio: "inherit",
      shell: true,
    })
  : spawn(npxCmd, ["-y", "@notionhq/notion-mcp-server"], {
      cwd: scriptDir,
      stdio: "inherit",
      shell: false,
    });

child.on("error", () => {
  process.stderr.write(
    "[notion-mcp] Missing npx. Install Node.js LTS (with npm/npx) or add npx to PATH.\n",
  );
  process.exit(1);
});

child.on("exit", (code) => {
  process.exit(code ?? 1);
});

File: .cursor/mcp/run_notion_mcp.py
#!/usr/bin/env python3
import os
import subprocess
import sys
from pathlib import Path
from shutil import which


def load_env_file(env_file: Path) -> None:
    if not env_file.exists():
        return
    for raw in env_file.read_text(encoding="utf-8").splitlines():
        line = raw.strip()
        if not line or line.startswith("#") or "=" not in line:
            continue
        key, value = line.split("=", 1)
        key = key.strip()
        value = value.strip().strip('"').strip("'")
        if key and key not in os.environ:
            os.environ[key] = value


def main() -> int:
    script_dir = Path(__file__).resolve().parent
    env_file = script_dir / "notion.env"
    load_env_file(env_file)

    token = os.environ.get("NOTION_TOKEN", "").strip()
    if not token:
        print(
            "[notion-mcp] Missing NOTION_TOKEN. Please copy .cursor/mcp/notion.env.example to .cursor/mcp/notion.env and fill it.",
            file=sys.stderr,
        )
        return 1

    npx_exe = which("npx")
    if not npx_exe:
        # Search PATH-relative candidates (cross-device safe — no hardcoded paths)
        import os
        for p in os.environ.get("PATH", "").split(os.pathsep):
            candidate = Path(p) / "npx.cmd"
            if candidate.exists():
                npx_exe = str(candidate)
                break

    if not npx_exe:
        print(
            "[notion-mcp] Missing npx. Install Node.js LTS (with npm/npx) or add npx to PATH.",
            file=sys.stderr,
        )
        return 1

    cmd = [npx_exe, "-y", "@notionhq/notion-mcp-server"]
    return subprocess.call(cmd, cwd=str(script_dir), shell=os.name == "nt")


if __name__ == "__main__":
    raise SystemExit(main())
